### PR TITLE
GitHub Integration update

### PR DESCRIPTION
**Pull request by ayman-m**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A Github Integration Update to Fetch Pull Requests as incidents and additional commands to pull the check run data of tasks that run on pull requests

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [x ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ x] No

## Must have
- [ ] Tests
- [x ] Documentation

### CODE REVIEW

**Review by DeanArbel — CHANGES_REQUESTED** [review]
Great contribution, I'm sure that'll be used by many users.
* Please add minor version release notes - see [this guide](https://xsoar.pan.dev/docs/integrations/release-notes) for reference.
* Please add a raw_result example of the new command under `content/Packs/GitHub/TestPlaybooks/test_data/check_run_response.json`.
* See the rest of my comments below.

File: Packs/GitHub/Integrations/GitHub/GitHub.py [hunk 1]
```diff
@@ -1251,6 +1314,28 @@ def fetch_incidents_command():
             if updated_at > last_time:
                 last_time = updated_at
 
+    if demisto.params().get('fetch_pull_requests'):
```
Comment (DeanArbel) [inline Packs/GitHub/Integrations/GitHub/GitHub.py:1317]: The integration should fetch either issues or pull requests, not both. Please separate the default issues fetch with an else block

File: Packs/GitHub/Integrations/GitHub/GitHub.py [hunk 2]
```diff
@@ -1224,6 +1224,69 @@ def get_github_actions_usage():
...
+    check_run_result = []
+
+    check_runs = list_check_runs(owner_name=owner_name, repository_name=repository_name, run_id=run_id, commit_id=commit_id)
+
+    for check_run in check_runs:
+            check_run_id = check_run.get('id', '')
+            check_external_id = check_run.get('external_id', '')
+            check_run_name = check_run.get('name', '')
+            check_run_app_name = check_run['app'].get('name', '')
+            check_run_pr = check_run['pull_requests']
```
Comment (DeanArbel) [inline Packs/GitHub/Integrations/GitHub/GitHub.py:1261]: Are `app` and `pull_requests` always expected to be in `check_run`? Might be better to use `.get(app, {})` and `.get('pull_requests')` instead

**Comment by DeanArbel** [comment]
Hi there @ayman-m if you are aiming to contribute this pack please address the relevant comments so we can get this pack merged, if not please let us know as soon as possible, as we are allocating resources to handle this contribution process.

File: Packs/GitHub/Integrations/GitHub/GitHub.py (hunk 1 above)
Comment (ayman-m) [reply Packs/GitHub/Integrations/GitHub/GitHub.py:1317]: @DeanArbel ,will push another commit to reflect that

File: Packs/GitHub/Integrations/GitHub/GitHub.py (hunk 2 above)
Comment (ayman-m) [reply Packs/GitHub/Integrations/GitHub/GitHub.py:1261]: @DeanArbel app and pull_requests are always there, but will use .get() anyways

**Comment by ayman-m** [comment]
> 
> 
> Hi there @ayman-m if you are aiming to contribute this pack please address the relevant comments so we can get this pack merged, if not please let us know as soon as possible, as we are allocating resources to handle this contribution process.

Hi @DeanArbel apologies for the delay of my response, will push a new commit to reflect the comments once I have a chance this week

**Comment by bakatzir** [comment]
@ayman-m When can we expect an update? We can also close this PR and re open when you'll have time. we want to keep the process fast.

**Pull request closed without merging**